test/e2e/kubectl/portforward.go

```diff
@@ -207,6 +264,51 @@ func runPortForward(ns, podName string, port int) *portForwardCommand {
 	}
 }
 
+// Reproduces issue #74551 (https://github.com/kubernetes/kubernetes/issues/74551)
+func doTestConnectionNeverReadRequestBody(ctx context.Context, f *framework.Framework) {
+	ginkgo.By("Creating the target pod")
+	pod := pfNeverReadRequestBodyPod()
+	if _, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+		framework.Failf("Couldn't create pod: %v", err)
+	}
+	if err := e2epod.WaitTimeoutForPodReadyInNamespace(ctx, f.ClientSet, pod.Name, f.Namespace.Name, framework.PodStartTimeout); err != nil {
+		framework.Failf("Pod did not start running: %v", err)
+	}
+
+	ginkgo.By("Running 'kubectl port-forward'")
+	cmd := runPortForward(f.Namespace.Name, pod.Name, 80)
+	defer cmd.Stop()
+
+	ginkgo.By("Requesting the local port")
+	for sentBodySize := 0; sentBodySize < 1024*1024*512; {
+		// try to mock a big request.
+		size := rand.Intn(4 * 1024 * 1024)
+		url := fmt.Sprintf("http://localhost:%d/header", cmd.port)
+		_, err := post(url, strings.NewReader(strings.Repeat("x", size)), nil)
+		if err != nil {
+			framework.Failf("post data to Pod: %v", err)
```

Comment:
I'm a little confused here... this test ensures the post completes without error, but ... isn't it posting to a pod that never reads the input? wouldn't we *expect* that to error, rather than silently completing successfully?